aderyn_core/src/detect/high/arithmetic_underflow_overflow.rs

```diff
@@ -0,0 +1,136 @@
+use std::{collections::BTreeMap, error::Error};
+
+use crate::{
+    ast::NodeID,
+    capture,
+    context::workspace_context::WorkspaceContext,
+    detect::detector::{IssueDetector, IssueDetectorNamePool, IssueSeverity},
+};
+use eyre::Result;
+use semver::{Op, VersionReq};
+
+#[derive(Default)]
+pub struct ArithmeticUnderflowOverflowDetector {
+    // Keys are source file name and line number
+    found_instances: BTreeMap<(String, usize, String), NodeID>,
+}
+
+fn version_req_below_0_8(version_req: &VersionReq) -> bool {
+    // Simplified logic to check if version_req is below 0.8
+    // Note: This might not cover all complex semver cases.
+    if version_req.comparators.len() == 1 {
+        let comparator = &version_req.comparators[0];
+        match comparator.op {
+            Op::Tilde | Op::Caret | Op::LessEq | Op::Greater | Op::GreaterEq | Op::Exact => {
+                return comparator.major == 0 && comparator.minor < Some(8);
+            }
+            Op::Less => {
+                return comparator.major == 0 && comparator.minor <= Some(8);
+            }
+            _ => {}
+        }
+    } else if version_req.comparators.len() == 2 {
+        let comparator_2 = &version_req.comparators[1];
+        match comparator_2.op {
+            Op::Less => {
+                return comparator_2.major == 0 && comparator_2.minor <= Some(8);
+            }
+            Op::LessEq | Op::Exact => {
+                return comparator_2.major == 0 && comparator_2.minor < Some(8);
+            }
+            _ => {}
+        }
+    }
+
+    false
+}
+
+impl IssueDetector for ArithmeticUnderflowOverflowDetector {
+    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
+        for pragma_directive in context.pragma_directives() {
```

Comment:
I see the problem. Here, you're retrieving all pragma directives, but later on line 67, you're retrieving all assignments separately. There's no link between the pragmas that you've checked and the assignments that you retrieve from the WorkspaceContext. Therefore, the pragma check becomes redundant.

To resolve:
1. Get all `source_units`
2. Find the pragma -> do the check. If the pragma is below 0.8.0...
3. Extract all the assignments from the source unit
4. Capture the unsafe math

Reply:
@alexroan Right indeed! Thanks! 🚀